system-test/spanner.ts

```diff
@@ -4876,6 +4877,42 @@ describe('Spanner', () => {
         queryStreamMode(done, PG_DATABASE, query, POSTGRESQL_EXPECTED_ROW);
       });
 
+      it('GOOGLE_STANDARD_SQL should execute mutation group using Batch write', function (done) {
```

Comment:
We don't seem to have any tests that cover the error handling of `batchWrite` (other than `Session not found`). We shoud add that? Either as an end-to-end test, or as a test with the Spanner mock server?

It would be good to have verification that:
1. The error handling works as expected if for example 1 mutation group fails, but the other X mutation groups in the request succeed. But also if there are multiple errors and multiple successes? Or only failures?
2. The events that are emitted by the stream that we return to the user are logical. E.g. does it only emit one `error` event and then stop? Or multiple `error` events if there are multiple errors? And what happens to the `done` event in those cases?

Reply:
I have added an end-to-end test for the error handling of `batchWrite`.
1. I have added a test for the case when there will be invalid mutation groups in the request. Whenever an invalid mutation group will get pass in the request, it will give the response in stream as 'data' event with an `error code of failure`.
2. We are able to get the error event in the case when we are passing an invalid list of mutation groups.
For example - An empty list of Mutation Group.